### PR TITLE
limit next and prev clicks

### PR DESCRIPTION
**Pull request by yudeikisorta**
Added 2 new properties to calendar: **limitPrevClicks** and **limitNextClicks** with integer values in order to limit Prev and Next navigation according to these parameters

### CODE REVIEW

File: src/Calendar.js [hunk 1]
```diff
@@ -883,6 +885,31 @@ function Calendar_constructor(element, overrides) {
 	}
 
 
+	function updateNavigationButtons() {
+	    var nextButtonLimit = parseInt(t.options.limitNextClicks, 10),
+            prevButtonLimit = parseInt(t.options.limitPrevClicks, 10);
```
Comment (criztovyl) [inline src/Calendar.js]: I think the default should be that the option is disabled, not limit to +-10.
As it's numeric I would suggest `-1`, meaning the limit is disabled.

File: src/Calendar.js [hunk 2]
```diff
@@ -883,6 +885,31 @@ function Calendar_constructor(element, overrides) {
...
+
+		//checking next button
+        if (nextButtonLimit >= 0) {
```
Comment (criztovyl) [inline src/Calendar.js]: You will need to check for `-1` here first. If `-1`, should be no-op.

File: src/Calendar.js [hunk 3]
```diff
@@ -883,6 +885,31 @@ function Calendar_constructor(element, overrides) {
...
+            if (navigationButtonClicks >= nextButtonLimit) {
+                toolbarsManager.proxyCall('disableButton', 'next');
+            } else {
+                toolbarsManager.proxyCall('enableButton', 'next');
+            }
+        }
+
+        //checking prev button
+        if (prevButtonLimit >= 0) {
```
Comment (criztovyl) [inline src/Calendar.js]: Same as above.

File: src/Calendar.js (hunk 1 above)
Comment (yudeikisorta) [inline src/Calendar.js]: 10 is not the default option, is the redix parameter for parseInt function 

https://www.w3schools.com/jsref/jsref_parseint.asp
The radix parameter is used to specify which numeral system to be used, for example, a radix of 16 (hexadecimal) indicates that the number in the string should be parsed from a hexadecimal number to a decimal number

by default the option is disabled unless its value is > -1

Comment (criztovyl) [inline src/Calendar.js]: I'm sorry, I missed the radix ._.

But how is it disabled when it's below 0?

**Comment by arshaw** [comment]
thanks @yudeikisorta , this is a nice hack, but i'd prefer to limit the allowed navigatable date range via explicit start/end dates or forward/backward date delta, rather than rely on "clicks" which might equate to a different jumped duration depending on the view.

more discussion here: https://github.com/fullcalendar/fullcalendar/issues/429

**Comment by arshaw** [comment]
i mentioned previously that I'd rather a date range limit the user's ability to navigate rather than a number of clicks.

It is now possible to restrict the date range: https://fullcalendar.io/docs/current_date/validRange/

So I will close this PR

**Pull request closed without merging**